### PR TITLE
Automatically generate the intra doc links

### PR DESCRIPTION
**Pull request by gkobeaga**
Addresses #245

The doc links are changed from `[comment](struct.StructName.html)` to `[comment](StructName)`.

I realized that the documentation for TweedieRegressor was not generating. To fix it and to match the public API of `linfa-linear` with the other crates (e.g. `KMeans` and `KMeansParams` from linfa_clustering), I made the following changes:
- The documentation for `TweedieRegressorValidParams` (which is not public), is moved for `TweedieRegressor`
- `TweedieRegressorParams` is now public

Small fixes:
- `FittedCountVectorizer` -> `CountVectorizer` in linfa-preprocessing

### CODE REVIEW

**Comment by YuhanLiin** [comment]
I think `TweedieRegressorValidParams` should also be made public.

Also, do you mind setting `#![deny(rustdoc::broken_intra_doc_links)]` for the crates?

**Comment by gkobeaga** [comment]
Instead of adding  `#![deny(rustdoc::broken_intra_doc_links)]` for the crates, how about checking the broken links in the CI? I tested it with [c00c9af](https://github.com/rust-ml/linfa/pull/246/commits/c00c9af82fe5e037f6772975a1cc8e2b8afc4674), and the CI is failing now for that commit.

I also added the `--workspace` flag to `cargo doc` in the CI to build the documents of all the crates.

`TweedieRegressorValidParams` is public now, as you suggested.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -14,13 +14,19 @@ jobs:
...
+
       - name: 'Build only' 
         uses: shalzz/zola-deploy-action@master
         env:
           BUILD_DIR: docs/website/
           TOKEN: ${{ secrets.TOKEN }}
           BUILD_ONLY: true
 
+      - name: Build Documentation
+        run: cargo doc --workspace --all --no-deps
```
Comment (YuhanLiin) [inline .github/workflows/docs.yml:26]: Delete the `-all` flag, which is redundant with the `--workspace` flag

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -34,7 +40,9 @@ jobs:
           components: rustfmt, rust-src
 
       - name: Build Documentation
-        run: cargo doc --all --no-deps
+        run: cargo doc --workspace --all --no-deps
```
Comment (YuhanLiin) [inline .github/workflows/docs.yml:43]: Here too

**Review by YuhanLiin — APPROVED** [review]

**Pull request merged**